src/pages/iou/ReceiptSelector/index.js

```diff
@@ -136,8 +167,95 @@ function ReceiptSelector(props) {
     ).current;
 
     return (
-        <View style={[styles.uploadReceiptView(isSmallScreenWidth)]}>
-            {!isDraggingOver ? (
+        <View style={[styles.flex1, !Browser.isMobile() && styles.uploadReceiptView(isSmallScreenWidth)]}>
+            {!isDraggingOver && Browser.isMobile() && (
```

Comment:
This becomes a wall of text...

How about...

```js
const buildPrimarySection= () => {
  if (Browser.isMobile()) {
    return buildMobilePrimarySection();
  } else {
    return buildDesktopPrimarySection();
  }
}

// ...

{!isDraggingOver && buildPrimarySection()|
```

This way, we highlight that this is a mobile/desktop switch, and one of them always renders.

The refactored fragments could be either local functions in this component (with no arguments), private functions in this module (with appropriate arguments introduced), or even new components, whichever you find preferable.

Reply:
@lukemorawski 

> made some amendments

Looks good! Please also check the other threads with minor comments.